### DISCUSSION TITLE
face clustering minimum

### DISCUSSION THREAD

**Discussion by 31337-4554551n** [nextcloud/recognize#1167] [Q&A]
```
Face recognition: 0 Queued files, Last classification: 12 hours ago, Scheduled background jobs: 0, 
```

```
 Face clustering: 42 faces left to cluster, Last clustering run: 12 hours ago, Scheduled background jobs: 0,
A minimum of 120 faces per user is necessary for clustering to kick in
```
If I'm reading this correctly, I have 78 pics with faces detected, but they won't be added into their correct people until I take another 42 photos with faces in them?

Is there any way to shrink those batches?
I definitely know I have photos with peoples faces that aren't showing up under people... and idk how to force em

[edit] to clarify I am using recognize to feed into memories

**Comment by 31337-4554551n** [comment]
Added 2 photos with faces 
" 43 faces left to cluster"... Confused

**Comment by marcelklehr** [comment]
120 faces *in total* are necessary, so if you already have some people populated then that's not the issue. It says `Scheduled background jobs: 0` so that's the issue. You can also run clustering manually by running `occ recognize:cluster-faces -b 5000` (batch size to limit memory consumption)

**Comment by 31337-4554551n** [comment]
Thank you, I did that, and am getting
```
ClusterDebug: Retrieving face detections for user blah
ClusterDebug: Not enough face detections found
```
for all the users.
 Is there a way I can find out why this isn't happening automatically/at all anymore? It did for a bunch to begin with, but I've added more photos since, and I also know it missed a few

**Reply by marcelklehr** [reply]
> Is there a way I can find out why this isn't happening automatically/at all anymore? It did for a bunch to begin with, but I've added more photos since, and I also know it missed a few

Yeah, that's strange. It seems like a bug. I'll look into this.

**Reply by 31337-4554551n** [reply]
Thank you, please let me know if I can give you any further info

**Reply by 31337-4554551n** [reply]
For more context, I added a bunch of photos today after a trip out.
Today I went for a trip with the girl. Took 8 photos. 7 of her, one with both of us. It's detected 6 of her, and added to her. But one of her is not detected, and the one with both of us isn't detected under either of us (all show up under memories timeline-today)

And the results as above, doesn't think there's any more work to do.

Would a recrawl do anything? Or if it didn't find it the first time, it wouldn't again. Could it break clustering or correctly recognized faces? Or is it completely unnecesary since it's telling me it's aware of 42 faces it just doesn't want to cluster them.

**Reply by 31337-4554551n** [reply]
> > Is there a way I can find out why this isn't happening automatically/at all anymore? It did for a bunch to begin with, but I've added more photos since, and I also know it missed a few
> 
> Yeah, that's strange. It seems like a bug. I'll look into this.

Hi, I was wondering if you've had a chance to look into this. Confirming it's still saying 
```
Face clustering: 42 faces left to cluster, Last clustering run: 8 hours ago, Scheduled background jobs: 0,
A minimum of 120 faces per user is necessary for clustering to kick in
```
And trying
```
occ recognize:cluster-faces -b 5000
```
Still gives me 
```
ClusterDebug: Retrieving face detections for user blah
ClusterDebug: Not enough face detections found
```

Though I've been using it this whole time, and more photos have been getting identified. But whatever those 42 are appear to be stuck

**Reply by marcelklehr** [reply]
Those 42 could be from a separate user. The 120 faces is per user. --  Can you perhaps open an issue for this instead of this discussion, then I'll be more probable to follow up :see_no_evil: sorry.

**Reply by 31337-4554551n** [reply]
Thank you again for your time. Issue opened

**Comment by 31337-4554551n** [comment]
Is there perhaps more info or logs I can provide for you to help you troubleshoot the issue?
Or anything I can do to force it to work, perhaps a recrawl? Or would that not be productive.

**Reply by marcelklehr** [reply]
A recrawl could help. Note, that it's normal that the face detection model does not find all faces, especially if you run in WASM mode.

**Comment by 31337-4554551n** [comment]
I'm also noticing *some* new photos are getting the proper treatment, and others aren't. Not seeing a pattern
